keras_nlp/metrics/perplexity.py

```diff
@@ -0,0 +1,142 @@
+# Copyright 2022 The KerasNLP Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Perplexity metric implementation based on `keras.metrics.Metric`."""
+
+import tensorflow as tf
+from tensorflow import keras
+
+
+class Perplexity(keras.metrics.Metric):
+    """Perplexity metric.
+    This class implements the perplexity metric. In short, this class calculates
+    the cross entropy loss and takes its exponent.
+    Note: This implementation is not suitable for fixed-size windows.
+    Args:
+        name: string. Name of the metric instance.
+        dtype: string or tf.dtypes.Dtype. Precision of metric computation. If
+               not specified, it defaults to tf.float32.
+        from_logits: bool. If True, `y_pred` (input to `update_state()`) should
+            be the logits as returned by the model. Otherwise, `y_pred` is a
+            tensor of probabilities.
+        pad_token_id: int. Token ID of the padding token. If provided, the mask
+            is computed by this class (all padding tokens are masked while
+            computing the cross entropy loss). Note that if this field is
+            provided, the `sample_weight` field in `update_state()` is ignored.
+        **kwargs: Other keyword arguments.
+    Examples:
+    ```python
+    # 1. update_state() and result()
+    perplexity = keras_nlp.metrics.Perplexity(name="perplexity")
+    target = tf.experimental.numpy.random.randint(low=0, high=10, size=(2, 5))
```

Comment:
Prefer using this: 
```
target = tf.random.uniform(shape=[2, 5],  maxval=10, dtype=tf.int32)
```

Since after the numpy stuff graduates from experimental namespace, this code would break. 

Additionally, we may want to format this example section like this: https://github.com/keras-team/keras-nlp/blob/master/keras_nlp/tokenizers/word_piece_tokenizer.py#L123, which is more clear on the output, and more testable.